lib/rubocop/cop/rails/bulk_change_table.rb

```diff
@@ -0,0 +1,269 @@
+          timestamp
+          time
+          date
+          binary
+          boolean
+          json
+          virtual
+          remove
+          change
+          timestamps
+          remove_timestamps
+        ].freeze
+
+        COMBINABLE_ALTER_METHODS = %i[
+          add_column
+          remove_column
+          remove_columns
+          change_column
+          add_timestamps
+          remove_timestamps
+        ].freeze
+
+        MYSQL_COMBINABLE_TRANSFORMATIONS = %i[
+          rename
+          index
+          remove_index
+        ].freeze
+
+        MYSQL_COMBINABLE_ALTER_METHODS = %i[
+          rename_column
+          add_index
+          remove_index
+        ].freeze
+
+        POSTGRESQL_COMBINABLE_TRANSFORMATIONS = %i[
+          change_default
+        ].freeze
+
+        POSTGRESQL_COMBINABLE_ALTER_METHODS = %i[
+          change_column_default
+          change_column_null
+        ].freeze
+
+        def on_def(node)
+          return unless support_bulk_alter?
+          return unless MIGRATIION_METHODS.include?(node.method_name)
+
+          recorder = AlterMethodsRecorder.new
+
+          node.body.each_child_node(:send) do |send_node|
+            if combinable_alter_methods.include?(send_node.method_name)
+              recorder.process(send_node)
+            else
+              recorder.flush
+            end
+          end
+
+          recorder.offensive_nodes.each { |n| add_offense_for_alter_methods(n) }
+        end
+
+        def on_send(node)
+          return unless support_bulk_alter?
+          return unless node.command?(:change_table)
+          return if include_bulk_options?(node)
+          return unless node.block_node
+          send_nodes = node.block_node.body.each_child_node(:send).to_a
+
+          transformations = send_nodes.select do |send_node|
+            combinable_transformations.include?(send_node.method_name)
+          end
+
+          add_offense_for_change_table(node) if transformations.size > 1
+        end
+
+        private
+
+        # @param node [RuboCop::AST::SendNode] (send nil? :change_table ...)
+        def include_bulk_options?(node)
+          # arguments: [(sym :table) (hash (pair (sym :bulk) _))]
+          options = node.arguments[1]
+          return false unless options
+          options.hash_type? &&
+            options.keys.any? { |key| key.sym_type? && key.value == :bulk }
+        end
+
+        def database
+          cop_config['Database'] || database_from_yaml
+        end
+
+        def database_from_yaml
+          return nil unless database_yaml
+          case database_yaml['adapter']
+          when 'mysql2'
+            MYSQL
+          when 'postgresql'
+            POSTGRESQL
+          end
+        end
+
+        def database_yaml
+          return nil unless File.exist?('config/database.yml')
+          yaml = YAML.load_file('config/database.yml')
+          return nil unless yaml.is_a? Hash
+          config = yaml['development']
```

Comment:
This Cop detect a database adapter when `adapter` is set as` development` in `config/database.yml`.
In other words, it cannot detect an adapter if using `DATABASE_URL`, other configuration file location and more. 
However, you can change it by setting `Database` with `.rubocop.yml`.